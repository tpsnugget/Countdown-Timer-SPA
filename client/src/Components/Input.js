import React from 'react'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'

const Input = props => {

    const { setTime, startTimer, time } = props

    const sendBackStartTimer = () => {
        startTimer()
    }

    const sendTimeToCountdownTimer = (e) => {
        setTime(e.target.value)
    }

    return (
        <Grid container justify='center' alignItems='center' spacing={5}>
            <Grid item xs={12} md={2}>
                <Typography variant='h4' data-testid='Input-countdown-text'>
                    Countdown:
                </Typography>
            </Grid>
            <Grid item xs={12} md={2}>
                <TextField data-testid='Input-textfield-input' label="(Min)" variant="outlined" onChange={sendTimeToCountdownTimer} >

                </TextField>
            </Grid>
            <Grid item xs={12} md={2}>
                {time >= 1 ?
                    <Button variant="contained" color="primary" >
                        <Typography variant='h4' >
                            START
                        </Typography>
                    </Button>
                    :
                    <Button variant="contained" data-testid='button' disabled onClick={sendBackStartTimer} >
                        <Typography variant='h4' >
                            START
                        </Typography>
                    </Button>

                }
            </Grid>
        </Grid>
    )
}

Input.propTypes = {
    setTime: PropTypes.func,
    startTimer: PropTypes.func,
    time: PropTypes.number
}

export default Input
