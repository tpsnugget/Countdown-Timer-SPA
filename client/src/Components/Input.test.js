import React from 'react'
import { render } from '@testing-library/react'
import Input from './Input'

test('<Input />', () => {
    const { container, getByTestId } = render(<Input />)

    // Take a snapshot
    expect(container.firstChild).toMatchSnapshot()

    // Get the Countdown: text
    const countDownText = getByTestId('Input-countdown-text')
    expect(countDownText.textContent).toBe('Countdown:')

    // Get the TextField input box
    const textField = getByTestId('Input-textfield-input')
    expect(textField.firstChild.firstChild.textContent).toBe('(Min)')

    // Get the button
    const submitButton = getByTestId('button')
    expect(submitButton).toHaveProperty('type', 'button')
})