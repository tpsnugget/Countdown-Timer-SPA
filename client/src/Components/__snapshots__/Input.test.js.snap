// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4"
      data-testid="Input-countdown-text"
    >
      Countdown:
    </h4>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
      data-testid="Input-textfield-input"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
      >
        (Min)
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              (Min)
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
      data-testid="button"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          START
        </h4>
      </span>
    </button>
  </div>
</div>
`;
